File: order.py
class Order:
    def __init__(self):
        self.items = []

    def add_item(self, item):
        self.items.append(item)
        print(f"Added: {item['name']}")

    def remove_item(self, item_name):
        for item in self.items:
            if item["name"].lower() == item_name.lower():
                self.items.remove(item)
                print(f"Removed: {item_name}")
                return

        print("Item not found in order.")

    def calculate_total(self):
        return sum(item["price"] for item in self.items)

    def show_order(self):
        if not self.items:
            print("Order is empty.")
            return

        print("\n--- CURRENT ORDER ---")

        for item in self.items:
            print(f"{item['name']} - ${item['price']}")

        print(f"\nTotal: ${self.calculate_total():.2f}")

File: order.c
#include <stdio.h>
#include <string.h>
#include "order.h"

void init_order(Order *order) {
    order->count = 0;
}

void add_item(Order *order, MenuItem item) {
    if (order->count < MAX_ITEMS) {
        order->items[order->count++] = item;
        printf("Added: %s\n", item.name);
    }
}

void remove_item(Order *order, char name[]) {
    for (int i = 0; i < order->count; i++) {

        if (strcmp(order->items[i].name, name) == 0) {

            for (int j = i; j < order->count - 1; j++) {
                order->items[j] = order->items[j + 1];
            }

            order->count--;
            printf("Removed: %s\n", name);
            return;
        }
    }

    printf("Item not found.\n");
}

float calculate_total(Order *order) {
    float total = 0;

    for (int i = 0; i < order->count; i++) {
        total += order->items[i].price;
    }

    return total;
}

void show_order(Order *order) {

    if (order->count == 0) {
        printf("Order is empty.\n");
        return;
    }

    printf("\n--- CURRENT ORDER ---\n");

    for (int i = 0; i < order->count; i++) {
        printf("%s - $%.2f\n",
               order->items[i].name,
               order->items[i].price);
    }

    printf("\nTotal: $%.2f\n",
           calculate_total(order));
}
